### ISSUE TITLE
Metar data not updating

### ISSUE THREAD

**Issue by lupis1111** [n0bel/PiClock3#4]
I’ve installed PiClock3 on a couple of different Pi’s per the directions in the readme file and I’m finding that the Metar data is not updating on either one.  Is that expected with the current state of the program.

**Comment by mtijrobinson** [comment]
I have this fixed as well as an issue when winds are calm not having a directional value. @n0bel I'd be happy to contribute more as I have time if you're still active on it. Great project.

In the meantime @lupis1111 the timer object just goes out of scope. Look at Metar.py line 231 in the current commit and add "self.timer" rather than "timer" and that will fix it.

**Comment by lupis1111** [comment]
Thanks @mtijrobinson, that fixed my issue.  Looking forward to updates to PiClock3, it’s a great program.

**Comment by SerBrynden** [comment]
Have you tried this fork of the original PiClock? https://github.com/SerBrynden/PiClock
It has more commits than PiClock3.  It's been updated to Python3 and PyQt5 to run on the latest Raspberry Pi OS and has everything working, such as radar and forecasts.  n0bel and I have been working on the original PiClock more so than PiClock3.

**Comment by mtijrobinson** [comment]
Awesome! I'll check that out. I like the modular plugin design of what he has going on here.

**Comment by SerBrynden** [comment]
Yeah, it's the next generation of PiClock, but hasn't seen much activity in awhile.  And I think n0bel has been absent for a few weeks too.

**Comment by n0bel** [comment]
The cause is understood, at least. The QTimer that refreshed METAR was assigned to a local variable, so Python collected it as soon as `start()` returned and it never fired once.

SerBrynden found it and fixed it in PR #8, which is now merged. I haven't tested it myself yet, so I'll leave this open until I have.

Thanks for the report, and sorry it sat this long.
